drivers/regulator/regulator_pca9420.c

```diff
@@ -538,7 +609,6 @@ static const struct regulator_driver_api api = {
 	.is_supported_voltage = regulator_pca9420_is_supported_voltage,
 	.set_voltage = regulator_pca9420_set_voltage,
 	.get_voltage = regulator_pca9420_get_voltage,
-	.set_current_limit = regulator_pca9420_set_current_limit,
```

Comment:
The problem is that according to the Linux API, the current limiting APIs are meant for current sinks that apply to outputs. In this case, things are a bit different. The current sink is at the input stage, and I guess it limits the output of the whole PMIC, so that $I_{OUT} = I_{BUCK1} + I_{BUCK2} + I_{LDO1} + I_{LDO2} + I_{CHARGER} + \ldots <= I_{IN}$. In the current implementation, the current reported is also incorrect because each supply has its limit, for example, LDO1 can only supply 1mA. Let me try to think how we can approach this, maybe we can have a small dedicated API for the “parent” regulator, where settings like this one can be configured at runtime.

Reply:
Sounds good. I think a small parent api might work here. Personally I would like to avoid recreating the regulator framework design Linux has, as I'm slightly concerned about overhead. Maybe the parent regulator device could implement these APIs, and the children could return ENOTSUP?